File: 1/multiples.cpp
#include <iostream>

using namespace std;

/*
 * adds up multiples of factor1 and factor2 up until the limit
 * ex: f1=3 f2=5 lim=10, sum == 23, factors are 3,5,6,9
*/
int multipleSum(const int& factor1, const int& factor2, const int& limit){
	int sum=0;
	for(int i = 1; i < limit; ++i){
		if(i%factor1 == 0 || i%factor2 == 0)
			sum+=i;
	}
	return sum;
}

int main(){
	cout << multipleSum(3,5,10) << endl;
	cout << multipleSum(3,5,1000) << endl;

	return 0;
}

File: 1/multiples.py
from time import *
print("""
    
Multiples of 3 and 5
Problem 1

  If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3,5,6 and 9. The sum of these multiples is 23.

  Find the sum of all the multiples of 3 or 5 below 1000.

  Answer: 233168

  """)
while True:
  factor1 = int(input("Enter first Factor: "))
  factor2 = int(input("Enter second Factor: "))
  limit = int(input("Enter limit: "))
  start=clock()
  multSet = set()
  for i in range(limit):
    if i%factor1 == 0 or i%factor2 == 0:
        multSet.add(i)
  multipleSum = sum(multSet)
  print("The multiple sum of {} and {} for numbers under {} is {}".format(factor1, factor2, limit, multipleSum))
  end = clock()
  print("Your program took {} seconds to run".format(end-start))
